sdk/python/tests/conftest.py

```diff
@@ -137,21 +140,33 @@ def simple_dataset_2() -> pd.DataFrame:
 @pytest.fixture(
     params=FULL_REPO_CONFIGS, scope="session", ids=[str(c) for c in FULL_REPO_CONFIGS]
 )
-def environment(request):
-    with construct_test_environment(request.param) as e:
-        yield e
+def environment(request, tmp_path_factory, worker_id):
```

Comment:
Ah yeah. Using pytest-xdist, each worker creates its own version of fixtures. I had a version of this PR that creates the fixture in a single worker and shares it across workers by pickling it, but that won't work because when a single fixture is used across workers, there's data corruption (because multiple processes write to the same registry location or sqlite online store).

`tmp_path_factory` is a holdover from there. Removing.